### PR TITLE
new: Add support for LKE node pool labels & taints

### PR DESCRIPTION
**Pull request by lgarber-akamai**
## 📝 Description

This pull request adds support for specifying the `labels` and `taints` field when creating, updating and accessing an LKE node pool.

Additionally, this pull request makes the following improvements to the object serialization logic:

* Adds a new `_flatten_request_body_recursive(...)` helper function to consolidate request body flattening logic and to address some minor inconsistencies
    * This also introduces support for serializing nested JSONObjects, which is a requirement for the `taints` field
* Breaks the `LKENodePool._populate(...)` method into multiple functions to improve modularity and readability

## ✔️ How to Test

The following test steps assume you have pulled down this PR locally and run `make install`.

### Unit Testing

```
make testunit
```

### Integration Testing

```
make TEST_COMMAND=models/lke
```

### Manual Testing

1. In a linode_api4-python sandbox environment (e.g. dx-devenv), run the following:

```python
import json
import os

from linode_api4 import LinodeClient, LKENodePoolTaint

client = LinodeClient(os.getenv("LINODE_TOKEN"))

# Create a cluster with labels and taints
cluster = client.lke.cluster_create(
    region="us-mia",
    label="test-cluster",
    kube_version="1.30",
    node_pools=[
        client.lke.node_pool(
            "g6-standard-1",
            2,
            labels={
                "foo.example.com/test": "bar",
                "foo.example.com/test2": "test",
            },
            taints=[
                LKENodePoolTaint(
                    key="foo.example.com/test", value="bar", effect="NoSchedule"
                )
            ],
        )
    ],
)

target_pool = cluster.pools[0]

print(f"Initial node pool:\n{json.dumps(target_pool._serialize(), indent=4)}\n")

# Update the labels and taints on this cluster
updated_labels = {
    "foo.example.com/test": "bar",
    "foo.example.com/test3": "cool",
}

updated_taints = [
    LKENodePoolTaint(key="foo.example.com/test3", value="cool", effect="NoExecute")
]

target_pool.labels = updated_labels
target_pool.taints = updated_taints

print(f"Updated pool (pre-refresh):\n{json.dumps(target_pool._serialize(), indent=4)}\n")

# Invalid the pool so it is implicitly refreshed
target_pool.invalidate()

print(f"Updated pool (post-refresh):\n{json.dumps(target_pool._serialize(), indent=4)}\n")
```

2. Ensure the output matches the following:

```json
Initial node pool:
{
    "count": 2,
    "autoscaler": {
        "enabled": false,
        "min": 2,
        "max": 2
    },
    "tags": [],
    "labels": {
        "foo.example.com/test": "bar",
        "foo.example.com/test2": "test"
    },
    "taints": [
        {
            "key": "foo.example.com/test",
            "value": "bar",
            "effect": "NoSchedule"
        }
    ]
}

Updated pool (pre-refresh):
{
    "count": 2,
    "autoscaler": {
        "enabled": false,
        "min": 2,
        "max": 2
    },
    "tags": [],
    "labels": {
        "foo.example.com/test": "bar",
        "foo.example.com/test3": "cool"
    },
    "taints": [
        {
            "key": "foo.example.com/test3",
            "value": "cool",
            "effect": "NoExecute"
        }
    ]
}

Updated pool (post-refresh):
{
    "count": 2,
    "autoscaler": {
        "enabled": false,
        "min": 2,
        "max": 2
    },
    "tags": [],
    "labels": {
        "foo.example.com/test": "bar",
        "foo.example.com/test2": "test"
    },
    "taints": [
        {
            "key": "foo.example.com/test",
            "value": "bar",
            "effect": "NoSchedule"
        }
    ]
}
```

### CODE REVIEW

File: linode_api4/groups/lke.py [hunk 1]
```diff
@@ -107,41 +107,22 @@ def cluster_create(
...
+                node_pools if isinstance(node_pools, list) else [node_pools]
             ),
+            "control_plane": control_plane,
         }
         params.update(kwargs)
 
-        result = self.client.post("/lke/clusters", data=drop_null_keys(params))
+        result = self.client.post(
+            "/lke/clusters",
+            data=_flatten_request_body_recursive(drop_null_keys(params)),
```
Comment (lgarber-akamai) [inline linode_api4/groups/lke.py:124]: We might be able to apply the same refactor in other group-level endpoint functions 👀

File: linode_api4/objects/base.py [hunk 2]
```diff
@@ -277,6 +277,8 @@ def save(self, force=True) -> bool:
                 ):
                     data[key] = None
 
+            # Ensure we serialize any values that may not be already serialized
+            data = _flatten_request_body_recursive(data)
```
Comment (lgarber-akamai) [inline linode_api4/objects/base.py:281]: User-defined unpopulated values were not being flattened here previously, causing an error in cases like this:

```python
pool = LKENodePool(client, 456, 123)
pool.taints = [LKENodePoolTaint(...)]
pool.save()
```

File: linode_api4/objects/lke.py [hunk 3]
```diff
@@ -29,6 +29,18 @@ class KubeVersion(Base):
...
+@dataclass
+class LKENodePoolTaint(JSONObject):
+    """
+    LKENodePoolTaint represents the structure of a single taint that can be
+    applied to a node pool.
+    """
+
+    key: str = ""
+    value: str = ""
+    effect: str = ""
```
[the remark this answers is not in the document]
Comment (lgarber-akamai) [reply linode_api4/objects/lke.py:41]: Good call!

**Review by ezilber-akamai — APPROVED** [review]
Implementation looks good. Unit, integration, and manual testing are all working properly. Nice work!

**Review by ykim-1 — APPROVED** [review]
LGTM, unit/integration tests pass locally. Nice work!

**Pull request merged**